vendor/engines/saas/lib/saas/engine.rb

```diff
@@ -17,6 +17,20 @@ class Engine < ::Rails::Engine
         })
       end
 
+      LoginController.class_eval do
+        before_action :page_job, only: [:public, :private]
+
+        :private
+
+        def page_job
+          auth_request = request.params['code'] && request.params['state']
+          if request.referer !~ /github\.com/ && !logged_in? && !auth_request
```

Comment:
True of theres already a token, referrer is needed for a login that is not yet authorized with GitHub, as the code and state are not yet available in request #2.